src/atlas/array/native/NativeDataStore.h

```diff
@@ -241,9 +241,49 @@ class DataStore : public ArrayDataStore {
 template <typename Value>
 class WrappedDataStore : public ArrayDataStore {
 public:
+    WrappedDataStore(Value* host_data, const ArraySpec& spec, size_t size): host_data_(host_data), size_(size),
```

Comment:
The `size` argument is redundant with `spec` and could lead to inconsistencies.
I propose to extract size from `spec` instead.

Reply:
I agree and am extracting `size` from `spec` now. See 6811aa37f. Thanks!